matchms/similarity/CosineGreedyNumba.py

```diff
@@ -0,0 +1,72 @@
+from typing import Tuple
+import numpy
+from matchms.similarity.collect_peak_pairs import collect_peak_pairs
+from matchms.typing import SpectrumType
+
+
+class CosineGreedyNumba:
+    """Calculate 'cosine similarity score' between two spectra.
+
+    The cosine score aims at quantifying the similarity between two mass spectra.
+    The score is calculated by finding best possible matches between peaks
+    of two spectra. Two peaks are considered a potential match if their
+    m/z ratios lie within the given 'tolerance'.
+    The underlying peak assignment problem is here sovled in a 'greedy' way.
+    This can perform notably faster, but does occationally deviate slightly from
+    a fully correct solution (as with the Hungarian algorithm). In practice this
+    will rarely affect similarity scores notably, in particular for smaller
+    tolerances.
+    """
+    def __init__(self, tolerance=0.3):
+        """
+        Args:
+        ----
+        tolerance: float
+            Peaks will be considered a match when <= tolerance appart. Default is 0.1.
+        """
+        self.tolerance = tolerance
+
+    def __call__(self, spectrum1: SpectrumType, spectrum2: SpectrumType) -> Tuple[float, int]:
+        """Calculate cosine score between two spectra.
+        Args:
+        ----
+        spectrum1: SpectrumType
+            Input spectrum 1.
+        spectrum2: SpectrumType
+            Input spectrum 2.
+        """
+        def get_peaks_arrays():
```

Comment:
get_peaks_arrays and get_matching_pairs duplicate code from Hungarian cosine (other PR). This can't be solved nicely now as the codes are in separate branches, but I will create an issue for it with a suggestion on how to solve this.

Reply:
Yes, I agree we should avoid those duplicates and work on it in an upcoming PR!